logback/src/main/java/com/linecorp/armeria/common/logback/RequestContextExportingAppender.java

```diff
@@ -194,83 +195,44 @@ public void setExport(String mdcKey) {
      * Use {@code add*()} methods instead.
      */
     public void setExports(String mdcKeys) {
-        ensureNotStarted();
         requireNonNull(mdcKeys, "mdcKeys");
-        Arrays.stream(mdcKeys.split(",")).map(String::trim).forEach(this::setExport);
+        builder.addKeyPattern(mdcKeys);
     }
 
     private void ensureNotStarted() {
-        if (exporter != null) {
+        if (isStarted()) {
             throw new IllegalStateException("can't update the export list once started");
         }
     }
 
     @Override
     protected void append(ILoggingEvent eventObject) {
-        final RequestContext ctx = RequestContext.currentOrNull();
-        if (ctx != null) {
-            final State state = state(ctx);
-            final RequestLog log = ctx.log();
-            final Set<RequestLogAvailability> availabilities = log.availabilities();
-
-            // Note: This equality check is extremely fast.
-            //       See RequestLogAvailabilitySet for more information.
-            if (!availabilities.equals(state.availabilities)) {
-                state.availabilities = availabilities;
-                export(state, ctx, log);
-            }
-
+        if (exporter == null) {
+            exporter = builder.build();
+        }
+        final Map<String, String> contextMap = exporter.export();
+        if (contextMap != null) {
             final Map<String, String> originalMdcMap = eventObject.getMDCPropertyMap();
             final Map<String, String> mdcMap;
 
             // Create a copy of 'state' to avoid the race between:
             // - the delegate appenders who iterate over the MDC map and
             // - this class who update 'state'.
```

Comment:
You need to move this comment block to the exporter.

Reply:
Oops, thanks!